File: challenge02.js
/**
 * @param {string[]} names - Array of names to frame
 * @returns {string} The framed names
 */

function createFrame(names) {
  let maxLength = Math.max(...names.map((name) => name.length));
  let border = "*".repeat(maxLength + 4);
  let result = [border];
  for (let name of names) {
    result.push("* " + name.padEnd(maxLength) + " *");
  }
  result.push(border);
  return result.join("\n");
}

File: challenge02.py
def create_frame(names: list[str]):
    """
    Creates a framed string containing a list of names.

    Args:
        names: A list of names to be framed.

    Returns:
        A string with the names framed by asterisks.
    """
    max_length: int = max(len(name) for name in names)
    border = "*" * (max_length + 4)
    result = [border]
    for name in names: 
        result.append("* " + name.ljust(max_length) + " *")
    result.append(border)
    return "\n".join(result)


